spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/dynamodb/it/DynamoDbTemplateIntegrationTest.java

```diff
@@ -0,0 +1,289 @@
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.enhanced.dynamodb.*;
+import software.amazon.awssdk.enhanced.dynamodb.model.PageIterable;
+import software.amazon.awssdk.enhanced.dynamodb.model.QueryConditional;
+import software.amazon.awssdk.enhanced.dynamodb.model.QueryEnhancedRequest;
+import software.amazon.awssdk.enhanced.dynamodb.model.ScanEnhancedRequest;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+/**
+ * Integration tests for DynamoDBTemplate.
+ *
+ * @author Matej Nedic
+ */
+
+@SpringBootTest(classes = { CredentialsProviderAutoConfiguration.class, RegionProviderAutoConfiguration.class,
+		AwsAutoConfiguration.class, DynamoDbAutoConfiguration.class })
+@Testcontainers
+class DynamoDbTemplateIntegrationTest {
+
+	private static final String REGION = "us-east-1";
+	private static DynamoDbTable dynamoDbTable;
+
+	@Autowired
+	private DynamoDbTemplate dynamoDbTemplate;
+
+	@DynamicPropertySource
+	static void snsProperties(DynamicPropertyRegistry registry) {
+		registry.add("spring.cloud.aws.dynamodb.region", REGION::toString);
+		registry.add("spring.cloud.aws.dynamodb.endpoint", localstack.getEndpointOverride(DYNAMODB)::toString);
+		registry.add("spring.cloud.aws.credentials.access-key", "noop"::toString);
+		registry.add("spring.cloud.aws.credentials.secret-key", "noop"::toString);
+		registry.add("spring.cloud.aws.region.static", "eu-west-1"::toString);
+	}
+
+	@Container
+	static LocalStackContainer localstack = new LocalStackContainer(
+			DockerImageName.parse("localstack/localstack:0.14.0")).withServices(DYNAMODB).withReuse(true);
+
+	@BeforeAll
+	public static void createTable() {
+		DynamoDbClient dynamoDbClient = DynamoDbClient.builder()
+				.endpointOverride(localstack.getEndpointOverride(DYNAMODB)).region(Region.of(localstack.getRegion()))
+				.credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("noop", "noop")))
+				.build();
+		dynamoDbTable = DynamoDbEnhancedClient.builder().dynamoDbClient(dynamoDbClient).build()
+				.table("more_complex_person", TableSchema.fromBean(MoreComplexPerson.class));
+		dynamoDbTable.createTable();
+	}
+
+	@Test
+	void dynamoDbTemplate_save_entitySuccessful() {
+		MoreComplexPerson moreComplexPerson = MoreComplexPerson.MoreComplexPersonBuilder.person().withName("foo")
+				.withLastName("bar").withUuid(UUID.randomUUID()).build();
+		assertThatCode(() -> dynamoDbTemplate.save(moreComplexPerson)).doesNotThrowAnyException();
```

Comment:
shouldn't we actually check if it has been persisted in dynamodb? "no exceptions" does not mean that it has been saved successfully. 

Or am i missing something?